ls1intum/Artemis
# `General`: Improve authorization checks for file download

`General`: Add authorization checks to access token generation for file download




### Checklist
#### General


#### Server
#### Client


### Motivation and Context


Fixes #761 

### Description

This change adds new, individual endpoints that generate access tokens for each requested resource (lecture attachment, attachment unit, file upload submission).
This is required in order to enforce a fine-grained access control on these resources.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 2 Lecture Attachments (one released, one unreleased)
- 2 Attachment Units (one released, one unreleased)
- 1 File Upload Submission


1. Login as an instructor 
2. You should be able to download all attachments and submissions
3. Login as a student
4. You should only be able to download the released attachments and your own file upload submission

#### Advanced Tests (requires you to send custom GET requests to the server; you could use [Postman](https://www.postman.com) for that):

1. Try to access a different students file upload submission
2. As a student try to access an unreleased lecture attachment or attachment unit
3. Try to get an access token for any  as a user not belonging to the course


### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| TokenProvider.java | 92% | 97% |
| FileResource.java | 80% | 93% |


### Screenshots
